pkg/vault/test/authorizer/TimelockAuthorizerPermissions.test.ts

```diff
@@ -0,0 +1,802 @@
+      });
+
+      it('emits an event', async () => {
+        const receipt = await authorizer.instance
+          .connect(getSender())
+          .scheduleGrantPermission(ACTION_1, user.address, WHERE_1, []);
+
+        expectEvent.inReceipt(await receipt.wait(), 'GrantPermissionScheduled', {
+          actionId: ACTION_1,
+          account: user.address,
+          where: WHERE_1,
+        });
+      });
+    }
+
+    context('when the sender is root', () => {
+      itScheduleGrantPermissionCorrectly(() => root);
+    });
+
+    context('when the sender is granter', () => {
+      sharedBeforeEach('makes a granter', async () => {
+        await authorizer.addGranter(ACTION_1, granter, EVERYWHERE, { from: root });
+        await authorizer.addGranter(ACTION_2, granter, EVERYWHERE, { from: root });
+      });
+      itScheduleGrantPermissionCorrectly(() => granter);
+    });
+  });
+
+  describe('revokePermission', () => {
+    const delay = DAY;
+
+    context('when there is a delay set to revoke permissions', () => {
+      sharedBeforeEach('set delay', async () => {
+        const setAuthorizerAction = await actionId(vault, 'setAuthorizer');
+        await authorizer.scheduleAndExecuteDelayChange(setAuthorizerAction, delay * 2, { from: root });
+        await authorizer.scheduleAndExecuteRevokeDelayChange(ACTION_1, delay, { from: root });
+      });
+
+      it('reverts if requires a schedule', async () => {
+        await expect(authorizer.revokePermission(ACTION_1, user, WHERE_1, { from: root })).to.be.revertedWith(
+          'REVOKE_MUST_BE_SCHEDULED'
+        );
+      });
+
+      it('can schedule a revoke permission', async () => {
+        const id = await authorizer.scheduleRevokePermission(ACTION_1, user, WHERE_1, [], { from: root });
+
+        // should not be able to execute before delay
+        await expect(authorizer.execute(id, { from: root })).to.be.revertedWith('ACTION_NOT_YET_EXECUTABLE');
+
+        await advanceTime(delay);
+        await authorizer.execute(id, { from: root });
+
+        expect(await authorizer.canPerform(ACTION_1, user, WHERE_1)).to.be.false;
+        expect(await authorizer.canPerform(ACTION_2, user, WHERE_2)).to.be.false;
+      });
+    });
+
+    context('when there is a no delay set to revoke permissions', () => {
+      it('reverts if the sender is not the revoker', async () => {
+        await expect(authorizer.revokePermission(ACTION_1, user, WHERE_1, { from: revoker })).to.be.revertedWith(
+          'SENDER_IS_NOT_REVOKER'
+        );
+      });
+
+      function itRevokesPermissionCorrectly(getSender: () => SignerWithAddress) {
+        context('when the user does not have the permission', () => {
+          it('ignores the request and cannot perform the requested action everywhere', async () => {
+            await expect(authorizer.revokePermission(ACTION_1, user, WHERE_1, { from: getSender() })).not.to.be
+              .reverted;
+
```

Comment:
How do you feel about this? Should we revert instead?

Reply:
I don't think it adds any value, can make things confusing, enables silent user error, and makes tests more complex.

Reply:
Created https://github.com/balancer/balancer-v2-monorepo/issues/2379